### DISCUSSION TITLE
JSON Editor Online 讓JSON變得好讀

### DISCUSSION THREAD

**Discussion by lx3211** [1101-GS4719-C/Python-Community-GS4719-C#560] [General]
前言
---
有時候上網抓資料
會碰到JSON格式的資料
直接看上去真的是慘不忍睹
舉例： [台北市Ubike即時服務資料.json](https://tcgbusfs.blob.core.windows.net/blobyoubike/YouBikeTP.json)
<img width="1433" alt="截圖 2022-04-25 10 32 45" src="https://user-images.githubusercontent.com/96198440/165011186-f3840f9c-ffc0-4f49-bb1c-444bf5f767d6.png">
如果要在程式碼中運用JSON的資料
還是得了解其中的結構
可是難道要一個一個括號慢慢對嗎？
除了在自己的IDE裡面下載相關套件之外
也可以使用**線上資源**
[JSON Editor Online](https://jsoneditoronline.org/#left=local.kiyuku&right=local.paruda)

如何使用JSON Editor Online
---
把資料複製丟進去的話會像下面那樣
<img width="1432" alt="截圖 2022-04-25 10 38 04" src="https://user-images.githubusercontent.com/96198440/165011602-31b54e8e-72a5-4e8a-8c46-6a658cb0234f.png">
這時候左上角有個**code | tree**的切換
切到tree那邊即可
Magic!
<img width="1434" alt="截圖 2022-04-25 10 39 42" src="https://user-images.githubusercontent.com/96198440/165011748-4798d6df-35be-4a77-ab19-0beafe3eae9d.png">
樹狀化JSON 哪一層有你要的資料都一目瞭然～
除此之外 隔壁也可以開另一個JSON檔案 非常方便ㄛ

**Comment by ThomasMeng0515** [comment]
看起來很方便很好用，如果是用python讀JSON檔案還要作資料的清理才能拿到想要的資訊，謝謝分享

**Comment by chia-kai-chang** [comment]
我是用這個工具：JSON Viewer http://jsonviewer.stack.hu/

**Comment by ghost** [comment]
json真的是一個又愛又恨的東西，多虧他很多東西都能encode成方便傳輸的格式，但在程式裡要使用他，又要調很多東西...

**Reply by chia-kai-chang** [reply]
在Python裡面使用JSON相當方便，把他當成Dict就可以了

**Comment by JSHT** [comment]
在網頁上看的話，自己都是用 [json formatter](https://chrome.google.com/webstore/detail/json-formatter/bcjindcccaagfpapjjmafapmmgkkhgoa?hl=zh-TW) 插件

**Comment by Ku-Hao-Chen** [comment]
謝謝分享，讓我知道json的使用方法。

**Comment by yucheng1028** [comment]
感謝分享，讓json閱讀容易很多

**Comment by minniewei** [comment]
我覺得這些東東非常重要 要不單看json真是太辛苦了

**Comment by bloodtrail0511** [comment]
json格式直接看真的是很痛苦XD有這個網站果然讀起來輕鬆不少

**Comment by awkward-willy** [comment]
線上的這些工具附帶顏色讓整體閱讀起來真的快樂不少
也可以透過 Vscode 裡的大綱或額外的延伸模組（如：Json Editor）達到更好的閱讀體驗

**Comment by jason79461385** [comment]
以前都直接看json檔案 雖然是看得懂啦 但就是會看到眼花 感謝分享

**Comment by minniewei** [comment]
json相當於其他語言真的比較好看懂，但還是覺得要有相關的工具

**Comment by Ben6uo** [comment]
謝謝分享 感覺在之後的大學生活會常用到

**Comment by Lynn12378** [comment]
真的好讀很多，資料太密集真的會看到眼睛不適，感謝分享
